src/platform/packages/shared/kbn-unified-tabs/src/components/tabbed_content/tabbed_content.tsx

```diff
@@ -143,7 +143,24 @@ export const TabbedContent: React.FC<TabbedContentProps> = ({
       maxItemsCount,
       onDuplicate: (item) => {
         const newItem = createItem();
-        newItem.label = `${item.label} (copy)`;
+        const baseLabel = item.label.includes('(copy)')
+          ? item.label.replace(/\s*\(copy\)( \d+)?$/, '')
+          : item.label;
+
+        // Find all existing copies to determine next number
+        const copyRegex = new RegExp(`^${baseLabel}\\s*\\(copy\\)( \\d+)?$`);
+        const copies = state.items
+          .filter((tab) => copyRegex.test(tab.label))
+          .map((tab) => {
+            const match = tab.label.match(/\(copy\)( (\d+))?$/);
+            return match && match[2] ? Number(match[2]) : 1; // match[2] is the number after (copy)
```

Comment:
```suggestion
            const match = tab.label.match(/\(copy\) (\d+)$/);
            return match && match[1] ? Number(match[1]) : 1; // match[1] is the number after (copy)
```
Nit: you might be able to simplify the capture groups here a bit.